server/client-connection.cpp

```diff
@@ -242,6 +210,24 @@ void ClientConnection::process_message(MessageIn& msg) {
 		EventSystem<ChatCommandEvent>::Get()->Emit(std::make_shared<ChatCommandEvent>(chat_command));
 		break;
 	}
+	case MessageType::LOGIN:
+	{
+		proto::UserLogin user_login;
+		user_login.ParseFromZeroCopyStream(&msg);
+		this->server->GetLuaSystem()->CallFunctions("onUserLogin", this, user_login.username());
+
+		this->server->SendWorld(shared_from_this());
+
+		this->OnJoinWorld();
+		this->user->AddEntityToWorld();
```

Comment:
This only handles the positive case, but all the other cases: Lua script doesn't exist, Lua throws an error, script wants to reject the user, user doesn't exist and script doesn't create it. All of those will likely just hard crash the server since the user variable isn't initialized otherwise, nor is it checked for null/unset.
I did see the Lua document note on this, but it still puts a heavy requirement on Lua to prevent a crash.

Reply:
Fixed by removing the onus of setting the user in the script